poetry.lock

```diff
@@ -0,0 +1,772 @@
+[[package]]
```

Comment:
i am no expert in poetry so i might need to read into this a bit. but is this lock needed to checked in or this can be generated locally?

Reply:
This lock is needed to be checked in, yes. It's a pattern that has been used by a number of different package managers from different platforms, like pip-tools, Cargo (Rust), npm (Node.JS) etc. The idea is that we have a file where we declare the dependencies and their constraints (which, in our case here, is `pyproject.toml`), and then have an automatically-generated "lock file", which is where those dependencies are pinned down to a specific version, to allow reproducible builds. And then every time we need to bump dependencies for some reason we just run `$ poetry update`, and then check in the lock file again with the newer versions.

Reply:
sounds great!